### PR TITLE
RHDEVDOCS-4470 - Added 1.7.0 release notes

### PR DESCRIPTION
**Pull request by DebarghoGhosh**
Aligned team: Dev Tools
OCP version for cherry-picking: enterprise- 4.11-4.12

JIRA issue:
https://issues.redhat.com/browse/RHDEVDOCS-4470

Preview pages: https://53541--docspreview.netlify.app/openshift-enterprise/latest/cicd/gitops/gitops-release-notes.html

SME+QE review: @iam-veeramalla @varshab1210 
Peer-review: @sounix000

### CODE REVIEW

File: modules/gitops-release-notes-1-7-0.adoc
Comment (varshab1210) [inline modules/gitops-release-notes-1-7-0.adoc:31]:
```suggestion
* With this update, to use the *Environments* feature on the *Developer* tab you will be forced to upgrade if you are on {gitops-title} lower than 1.7 and {product-title} higher than or on 4.15. link:https://issues.redhat.com/browse/GITOPS-2415[GITOPS-2415] 
```

File: modules/gitops-release-notes-1-7-0.adoc
Comment (varshab1210) [inline modules/gitops-release-notes-1-7-0.adoc:52]:
```suggestion
* Before this update, operator was not reconciling **mountsatoken** and **serviceaccount** settings for the repository server. While this has been fixed link:https://issues.redhat.com/browse/GITOPS-1873[GITOPS-1873], deletion of **serviceaccount** does not revert the ServiceAccount to **default**. A workaround for this is manually setting spec.repo.serviceaccount to **default** ServiceAccount.
```

cc: @wtam2018 @iam-veeramalla please suggest

**Comment by varshab1210** [comment]
Please update the **Compatibility and support matrix** as well

File: modules/gitops-release-notes-1-7-0.adoc
Comment (iam-veeramalla) [inline modules/gitops-release-notes-1-7-0.adoc:47]: Please mention that this feature is a technical preview feature which is inline with our upstream Argo CD.

Comment (iam-veeramalla) [inline modules/gitops-release-notes-1-7-0.adoc:47]: This is a feature not fix

**Review by iam-veeramalla — CHANGES_REQUESTED** [review]
@DebarghoGhosh PTAL

**Comment by iam-veeramalla** [comment]
@DebarghoGhosh This release updates the Argo CD, Helm and Kustomize versions. Please make sure it is reflected in the docs.

You can refer to the versions here.
https://spaces.redhat.com/pages/viewpage.action?spaceKey=GITOPS&title=GitOps+Component+Matrix

File: modules/go-compatibility-and-support-matrix.adoc
Comment (varshab1210) [inline modules/go-compatibility-and-support-matrix.adoc:18]:
```suggestion
|1.7.0    |0.0.46 TP |3.10.0 GA|4.5.7 GA   |2.5.4 GA |2.4.5 GA       |2.35.1 GA |7.5.1 GA |2.4.5 TP|4.8-4.11
```

**Comment by iam-veeramalla** [comment]
LGTM

File: modules/go-compatibility-and-support-matrix.adoc
Comment (sounix000) [inline modules/go-compatibility-and-support-matrix.adoc:17]: Change `**Notifications Controller**` to `*Notifications Controller*`. AsciiDoc does not need double asterisks for bold font. Also, there will be similarity with the rest of the table headers.

Comment (sounix000) [inline modules/go-compatibility-and-support-matrix.adoc:17]: Change `**Notifications Controller**` to `*Notifications Controller*`. Two reasons:

- AsciiDoc does not need double asterisks to render bold font.
- Single asterisks will maintain consistency with the rest of the table headers.

File: modules/gitops-release-notes-1-7-0.adoc
Comment (sounix000) [inline modules/gitops-release-notes-1-7-0.adoc:16]: I think it will be `environment variables`. Please double-check with the SMEs.

File: modules/gitops-release-notes-1-7-0.adoc
Comment (sounix000) [inline modules/gitops-release-notes-1-7-0.adoc:18]: Suggested version:

* With this update, the default `nodeSelector "kubernetes.io/os": "linux"` key-value pair is added to all workloads such that they only schedule on Linux nodes. In addition, any custom node selectors are added to the default and take precedence if they have the same key.

File: modules/gitops-release-notes-1-7-0.adoc
Comment (sounix000) [inline modules/gitops-release-notes-1-7-0.adoc:23]: Suggested version:

* With this update, you can use the RBAC policy matcher mode to select from the following options: `glob` (default) and `regex`.

File: modules/gitops-release-notes-1-7-0.adoc
Comment (sounix000) [inline modules/gitops-release-notes-1-7-0.adoc:25]: Suggested version:

* With this update, you can customize resource behavior using the following additional subkeys:
+
|===
| *Subkey* | *Key form* | *Mapped field in `argocd-cm`*
| `resourceHealthChecks` | `resource.customizations.health.<group_kind>` | `resource.customizations.health`
| `resourceIgnoreDifferences` | `resource.customizations.ignoreDifferences.<group_kind>` | `resource.customizations.ignoreDifferences`
| `resourceActions` | `resource.customizations.actions.<group_kind>` | `resource.customizations.actions`
|===
+
[NOTE]
====
In future releases, there is a possibility to deprecate the old method of customizing resource behavior by using only `resourceCustomization` and not subkeys.
====

File: modules/gitops-release-notes-1-7-0.adoc
Comment (sounix000) [inline modules/gitops-release-notes-1-7-0.adoc:31]: The wordings around the versions seems to be complicated. Can we simplify it?

File: modules/gitops-release-notes-1-7-0.adoc
Comment (sounix000) [inline modules/gitops-release-notes-1-7-0.adoc:33]: I suggest breaking down the sentence for better readability.

File: modules/gitops-release-notes-1-7-0.adoc
Comment (sounix000) [inline modules/gitops-release-notes-1-7-0.adoc:46]: Suggested version:

* Before this update, {gitops-title} releases were affected by an issue of dex pods failing with `CreateContainerConfigError` error when the `anyuid` SCC was assigned to the dex service account`. This update fixes the issue by assigning a default user id to the dex container.

File: modules/gitops-release-notes-1-7-0.adoc
Comment (sounix000) [inline modules/gitops-release-notes-1-7-0.adoc:55]: Suggested version:

* Before this update, the Operator did not reconcile the **mountsatoken** and `ServiceAccount` settings for the repository server. While this has been fixed, deletion of the service account does not revert to default.
+
Workaround: Manually set the ``spec.repo.serviceaccount` field to the `default` service account.

File: modules/gitops-release-notes-1-7-0.adoc
Comment (sounix000) [inline modules/gitops-release-notes-1-7-0.adoc:50]: Suggested version:

* Before this update, the application controllers restarted multiple times due to the unresponsiveness of liveness probes. This update fixes the issue by removing the liveness probe in the `statefulset` application controller.

File: modules/gitops-release-notes-1-7-0.adoc
Comment (sounix000) [inline modules/gitops-release-notes-1-7-0.adoc:48]: Suggested version:

* Before this update, {gitops-title} used the RHSSO (KeyCloak) through OIDC in addition to Dex. However, with a recent security fix, the certificate of RHSSO could not be validated when configured with a certificate not signed by one of the well known certificate authorities. This update fixes the issue; you can provide a custom certificate to verify the KeyCloak's TLS certificate while communicating with it. In addition, you can add `rootCA` to their Argo CD custom resource `.spec.keycloak.rootCA` field. The Operator reconciles such changes and updates the `oidc.config` in `argocd-cm` config map with the PEM encoded root certificate.

**Review by sounix000 — COMMENTED** [review]
Approving in advance, with the expectation that the author will make the suggested modifications.

**Review by sounix000 — APPROVED** [review]

File: modules/gitops-release-notes-1-7-0.adoc
Comment (bburt-rh) [inline modules/gitops-release-notes-1-7-0.adoc:9]:
```suggestion
{gitops-title} 1.7.0 is now available on {product-title} 4.8, 4.9, 4.10, and 4.11.
```
Per ISG: Use the Oxford comma for items in a series.

File: modules/gitops-release-notes-1-7-0.adoc
Comment (bburt-rh) [inline modules/gitops-release-notes-1-7-0.adoc:21]:
```suggestion
```

File: modules/gitops-release-notes-1-7-0.adoc
Comment (bburt-rh) [inline modules/gitops-release-notes-1-7-0.adoc:26]:
```suggestion
+
```
The table should be aligned with the previous bulleted item

File: modules/gitops-release-notes-1-7-0.adoc
Comment (bburt-rh) [inline modules/gitops-release-notes-1-7-0.adoc:44]:
```suggestion
+
```
The TP admonition needs to align with the previous bulleted item

File: modules/gitops-release-notes-1-7-0.adoc
Comment (bburt-rh) [inline modules/gitops-release-notes-1-7-0.adoc:36]: There's a rendering issue in the preview where the `+` signs are displaying. Maybe adding the `+` in line 26 will fix it.

File: modules/gitops-release-notes-1-7-0.adoc
Comment (bburt-rh) [inline modules/gitops-release-notes-1-7-0.adoc:48]:
```suggestion
```

File: modules/gitops-release-notes-1-7-0.adoc
Comment (bburt-rh) [inline modules/gitops-release-notes-1-7-0.adoc:56]:
```suggestion
* Before this update, {gitops-title} used the RHSSO (Keycloak) through OIDC in addition to Dex. However, with a recent security fix, the certificate of RHSSO could not be validated when configured with a certificate not signed by one of the well known certificate authorities. This update fixes the issue; you can provide a custom certificate to verify the KeyCloak's TLS certificate while communicating with it. In addition, you can add `rootCA` to the Argo CD custom resource `.spec.keycloak.rootCA` field. The Operator reconciles such changes and updates the `oidc.config in argocd-cm` config map with the PEM encoded root certificate. link:https://issues.redhat.com/browse/GITOPS-2214[GITOPS-2214]
```
- not sure what `their` refers to. Should it be `the`?
- Looks like there's an extra back tick that should be omitted.

File: modules/gitops-release-notes-1-7-0.adoc
Comment (bburt-rh) [inline modules/gitops-release-notes-1-7-0.adoc:54]: Should `dex` be capitalized here? You capitalize it in the next item.

File: modules/gitops-release-notes-1-7-0.adoc
Comment (bburt-rh) [inline modules/gitops-release-notes-1-7-0.adoc:63]:
```suggestion
* Before this update, the Operator did not reconcile the `mountsatoken` and `ServiceAccount` settings for the repository server. While this has been fixed, deletion of the service account does not revert to default. link:https://issues.redhat.com/browse/GITOPS-1873[GITOPS-1873]
```

Should `ServiceAccount` be in back ticks? Or edit to say `service account`?

File: modules/gitops-release-notes-1-7-0.adoc
Comment (bburt-rh) [inline modules/gitops-release-notes-1-7-0.adoc:41]:
```suggestion
* With this update, to use the *Environments* feature on the *Developer* tab you must upgrade if you are using a {gitops-title} version prior to 1.7 and {product-title} 4.15 or above. link:https://issues.redhat.com/browse/GITOPS-2415[GITOPS-2415] 
```
- consider replacing `will be forced to` with `must`
- rewrite awkward syntax referring to version prerequisites

File: modules/gitops-release-notes-1-7-0.adoc
Comment (bburt-rh) [inline modules/gitops-release-notes-1-7-0.adoc:27]:
```suggestion
[options=header]
|===
```
Format the first row of the table as a header.

**Comment by bburt-rh** [comment]
@DebarghoGhosh Before I merge this PR, you need to fix the issues noted in my line edits and comments. 

image

cc @sounix000

File: modules/gitops-release-notes-1-7-0.adoc
Comment (iam-veeramalla) [reply modules/gitops-release-notes-1-7-0.adoc:56]: @DebarghoGhosh can you please add the below 

An example Argo CD with Keycloak configuration should look something below

```
apiVersion: argoproj.io/v1alpha1
kind: ArgoCD
metadata:
  name: example-argocd
spec:
  sso:
    keycloak:
      rootCA: ''
    provider: keycloak
.......
.......
```

**Review by iam-veeramalla — CHANGES_REQUESTED** [review]
PTAL  https://github.com/openshift/openshift-docs/pull/53541#discussion_r1049155256


Sorry for the last minute update.

File: modules/gitops-release-notes-1-7-0.adoc
Comment (DebarghoGhosh) [reply modules/gitops-release-notes-1-7-0.adoc:63]: back ticks

File: modules/gitops-release-notes-1-7-0.adoc
Comment (DebarghoGhosh) [reply modules/gitops-release-notes-1-7-0.adoc:54]: capitalized

**Pull request merged**
